File: app/src/main/java/com/babestudios/companyinfouk/data/model/search/SearchHistoryItem.java
package com.babestudios.companyinfouk.data.model.search;


import java.util.Objects;

public class SearchHistoryItem {

	public String companyName;
	public String companyNumber;
	private long searchTime;


	public SearchHistoryItem(String companyName, String companyNumber, long searchTime) {
		this.companyName = companyName;
		this.companyNumber = companyNumber;
		this.searchTime = searchTime;
	}

	@Override
	public boolean equals(Object searchItem){
		if (searchItem == null) {
			return false;
		}
		if (!SearchHistoryItem.class.isAssignableFrom(searchItem.getClass())) {
			return false;
		}
		final SearchHistoryItem other = (SearchHistoryItem) searchItem;
		return Objects.equals(this.companyNumber, other.companyNumber);
	}

}

File: app/src/main/java/com/babestudios/companyinfouk/data/model/search/SearchHistoryItem.kt
package com.babestudios.companyinfouk.data.model.search


import java.util.Objects

class SearchHistoryItem(var companyName: String, var companyNumber: String, private val searchTime: Long) {

	override fun equals(searchItem: Any?): Boolean {
		if (searchItem == null) {
			return false
		}
		if (!SearchHistoryItem::class.java.isAssignableFrom(searchItem.javaClass)) {
			return false
		}
		val other = searchItem as SearchHistoryItem?
		return this.companyNumber == other!!.companyNumber
	}

}
